borg/helpers.py

```diff
@@ -227,10 +233,22 @@ class IncludePattern:
     path match as well.  A trailing slash makes no difference.
     """
     def __init__(self, pattern):
-        self.pattern = os.path.normpath(pattern).rstrip(os.path.sep)+os.path.sep
+        def match(path):
+            return (path+os.path.sep).startswith(self.pattern)
+
+        # HFS+ converts paths to a canonical form, so users shouldn't be
+        # required to enter an exact match
+        if sys.platform in ('darwin',):
+            # repository paths will be mostly in NFD, as the OSX exception list
+            # to NFD is small, so normalize to that form for best performance
+            pattern = unicodedata.normalize("NFD", pattern)
+            self.match = lambda p: match(unicodedata.normalize("NFD", p))
```

Comment:
maybe there could be just 2 match functions get defined above: match and match_normalized and here, we would just normalize the pattern and assign self.match = match_normalized.